### ISSUE TITLE
Feat: keybinding states

### ISSUE THREAD

**Issue by Macavirus** [rabite0/hunter#69]
Feature request while you're working out custom key bindings:

Would it be possible to do key "arpeggios" as per ranger? Ideally would support more than one in-between state.

Use case: I often find it useful to do multiple types of file sorting that dont' really deserve top-level keybindings. For instance, in `ranger` I may use `o s` to sort by size, `o c` to sort by ctime, `o t` to sort by type. Fitting all of these under one generic sort keybinding is very useful.

Also for inspiration, emacs hydra mode: https://github.com/abo-abo/hydra

**Comment by rabite0** [comment]
Yes, I've been thinking about adding some kind of (persistent?) macro recording and key layering system. Currently working on fixing the most glaring issues before I add yet more features, but it'll come.

**Comment by Macavirus** [comment]
:+1: Looking forward to it. Thanks for hunter, it is really promising!
